soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/SoQLError.scala

```diff
@@ -100,14 +128,15 @@ object TableFinderError {
         for {
           fields <- data[Fields](v)
           source <- sourceOpt[RNS](v)
-          position <- position(v)
         } yield {
-          PermissionDenied(source, position, fields.name)
+          PermissionDenied(source, fields.name)
         }
     }
   }
 
-  case class RecursiveQuery[+RNS](source: Option[ScopedResourceName[RNS]], position: Position, stack: Seq[CanonicalName]) extends TableFinderError[RNS]
+  case class RecursiveQuery[+RNS](definiteSource: Source[RNS], stack: Seq[CanonicalName]) extends TableFinderLookupError[RNS] {
```

Comment:
Hm. Makes me wonder if the `CallerStack` concept should be generalized so it can be used here, too. 🤷

Reply:
Definitely not; this is an error, and its responsibility is saying "you had a recursive  query, the chain was a -> b -> c -> b oops (or whatever)" so all it needs to know is that sequence of canonical-names; the whole CallerStack is just that exact list but with each node annotated with some extra stuff the tablefinder cares about.

Reply:
test